src/java/voldemort/server/niosocket/AsyncRequestHandler.java

```diff
@@ -66,9 +71,63 @@ public AsyncRequestHandler(Selector selector,
                                RequestHandlerFactory requestHandlerFactory,
                                int socketBufferSize,
                                NioSelectorManagerStats nioStats) {
-        super(selector, socketChannel, socketBufferSize, nioStats.getServerCommBufferStats());
+        super(selector, socketChannel, socketBufferSize);
         this.requestHandlerFactory = requestHandlerFactory;
         this.nioStats = nioStats;
+
+        initializeStreams(socketBufferSize, nioStats.getServerCommBufferStats());
+        if(this.inputStream == null || this.outputStream == null) {
+            throw new VoldemortApplicationException("InputStream or OuputStream is null after initialization");
+        }
+    }
+
+    /**
+     * Flips the output buffer, and lets the Selector know we're ready to write.
+     * 
+     * @param selectionKey
+     */
+
+    protected void prepForWrite(SelectionKey selectionKey) {
```

Comment:
Copy/Paste mistake, it was Protected inside SelectorManager, it can be private now. Will fix it.